client/sites-dashboard-v2/site-preview-pane/dotcom-preview-pane.tsx

```diff
@@ -104,6 +105,14 @@ const DotcomPreviewPane = ( {
 				setSelectedSiteFeature,
 				selectedSiteFeaturePreview
 			),
+			createFeaturePreview(
+				DOTCOM_STAGING_SITE,
+				__( 'Staging Site' ),
+				isAtomicSite && ! isPlanExpired,
```

Comment:
Looks like we need to check for `siteHasFeature( state, siteId, FEATURE_SITE_STAGING_SITES );` here; this might be why @miksansegundo was seeing the error.

Reply:
I used the store sandbox for those sites, and their plan appears as Free, not Creator.

<img width="272" alt="Screenshot 2567-05-30 at 17 39 34" src="https://github.com/Automattic/wp-calypso/assets/1881481/0356187a-177f-4fec-ad76-b32f4e79dcf8">

Reply:
Just confirmed that it works fine when I enable the store sandbox.

Reply:
I was following the original logic

https://github.com/Automattic/wp-calypso/blob/92f9ef4015cf4fc6a9e3799d4e5c2df9001ebe0c/client/my-sites/hosting/main.js#L207-L212

not sure if the condition needs to be replicated here 🤔

Reply:
hmmm... the issue is the free plan is not an expired plan, so you can see the tab here. I'll add the feature check to show the upsell banner!

Reply:
But if `isAtomicSite` is true then it can't be Free?

Reply:
Added the upsell nudge by https://github.com/Automattic/wp-calypso/pull/91276/commits/e3f81f98d312e7b43c6c110a0195d5746ee4bc4d

<img width="700" alt="image" src="https://github.com/Automattic/wp-calypso/assets/13596067/e0334288-549f-46be-8e9c-6ccdc276391f">

Reply:
> But if isAtomicSite is true then it can't be Free?

We won't revert the atomic site to the simple site after the plan has expired and changed to free. I'm not very sure as it's my test site. Maybe it's not a real case 🤔